### PR TITLE
Move command line option definitions into a dedicated file

### PR DESCRIPTION
**Pull request by 0dvictor**
config.rs has reached the 3000 line tidy limit, this commit moves command line option definitions into a new file - options.rs,  and leaves the rest of configuration infrastructure in config.rs.

### CODE REVIEW

**Comment by 0dvictor** [comment]
r? @Centril 
cc @tmandry

**Comment by 0dvictor** [comment]
Based on discussions in #67195, this PR is the first step refactoring `config.rs`.
Other functions that could be moved out is `fn build_session_options` and related ones; however, there are different opinions on whether `Option` definitions and building logic should stay in the same file. We probably can discuss further on which file should `fn build_session_options` reside while reviewing this PR.

File: src/librustc_session/options.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+// incremental compilation cache before proceeding.
+//
+// [UNTRACKED]
+// Incremental compilation is not influenced by this option.
+//
+// If you add a new option to this struct or one of the sub-structs like
+// `CodegenOptions`, think about how it influences incremental compilation. If in
+// doubt, specify [TRACKED], which is always "correct" but might lead to
+// unnecessary re-compilation.
+top_level_options!(
```
Comment (Centril) [inline src/librustc_session/options.rs]: Let's move the macros as well (this one and `options!`)?

Comment (0dvictor) [reply src/librustc_session/options.rs]: Sure, updated.

**Comment by Centril** [comment]
Thanks; r=me with rebase

**Comment by tmandry** [comment]
@bors delegate=0dvictor

(Use @ bors r=Centril once rebased)

**Comment by 0dvictor** [comment]
The PR has now been rebased.

@bors r=Centril

**Pull request merged**